cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -390,14 +503,20 @@ impl LockedManifestCatalog {
         manifest: &TypedManifestCatalog,
         seed_lockfile: Option<&LockedManifestCatalog>,
         client: &impl catalog::ClientTrait,
+        flake_locking: &impl LockFlakeInstallableTrait,
     ) -> Result<LockedManifestCatalog, LockedManifestError> {
-        let groups = Self::collect_package_groups(manifest, seed_lockfile)?;
+        let catalog_groups = Self::collect_package_groups(manifest, seed_lockfile)?;
```

Comment:
nonblocking: seems like this might be simpler if we first split into `Vec<LockedPackage::Flake>` and `Vec<LockedPackage::Catalog>`. That could let us completely drop e.g. `ManifestPackageDescriptor::invalidates_existing_resolution`

Reply:
we'd still need `invalidates_existing_resolution` on each subtype then tho.

We can change the `collect_*` functions to take `Vec<LockedPackage::*>` and `Allow`, but that also comes with a lot of churn in the tests.
since this pr is already bigger than i expected it to become we can maybe push that to a follow up/greenkeeping issue.